protocol_decoder/src/lib.rs

```diff
@@ -1,6 +1,10 @@
 #![feature(linked_list_cursors)]
 #![feature(trait_alias)]
 #![feature(iter_array_chunks)]
+// TODO: address these lints
+#![allow(unused)]
+#![allow(clippy::type_complexity)]
+#![allow(private_interfaces)]
```

Comment:
I'm good with these for now, but yeah, I can probably clean these up in like an hour.

Reply:
Haha no rush, I can take care of them as well 🙂